centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/hooks/useLinkToResourceStatus.ts

```diff
@@ -0,0 +1,39 @@
+import { equals, includes } from 'ramda';
+
+import {
+  getResourcesUrlForMetricsWidgets,
+  getResourcesUrlForStatusGrid,
+  resourceBasedWidgets
+} from '../utils';
+
+interface UseLinkToResourceStatus {
+  getLinkToResourceStatusPage: (data, name, options) => string;
+}
+
+const useLinkToResourceStatus = (): UseLinkToResourceStatus => {
+  const getLinkToResourceStatusPage = (data, name, options): string => {
+    if (!includes(name, resourceBasedWidgets)) {
+      return '';
+    }
+
+    if (equals(name, 'centreon-widget-statusgrid')) {
```

Comment:
I would change the condition to `options?.statusses && options?.states && data?.resources` rather than explictly name 'statusgrid' because another widget with the same options type might use this condition